flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManagerTest.java

```diff
@@ -470,6 +474,43 @@ private void testNotificationAboutNotEnoughResources(boolean withNotificationGra
         };
     }
 
+    /**
+     * Test that checkResourceRequirements will only be triggered once after multiple trigger
+     * function calls.
+     */
+    @Test
+    public void testRequirementCheckOnlyTriggeredOnce() throws Exception {
+        new Context() {
+            {
+                final AtomicInteger checkTimes = new AtomicInteger();
+                resourceAllocationStrategyBuilder.setTryFulfillRequirementsFunction(
+                        (ignored1, ignored2) -> {
+                            checkTimes.addAndGet(1);
+                            return ResourceAllocationResult.builder().build();
+                        });
+                runTest(
+                        () -> {
+                            final ResourceRequirements resourceRequirements1 =
+                                    createResourceRequirementsForSingleSlot();
+                            final ResourceRequirements resourceRequirements2 =
+                                    createResourceRequirementsForSingleSlot();
+                            final TaskExecutorConnection taskExecutionConnection =
+                                    createTaskExecutorConnection();
+                            getSlotManager().processResourceRequirements(resourceRequirements1);
+                            getSlotManager().processResourceRequirements(resourceRequirements2);
+                            getSlotManager()
+                                    .registerTaskManager(
+                                            taskExecutionConnection,
+                                            new SlotReport(),
+                                            DEFAULT_TOTAL_RESOURCE_PROFILE,
+                                            DEFAULT_SLOT_RESOURCE_PROFILE);
+                            getScheduledExecutor().triggerNonPeriodicScheduledTasks();
+                            assertThat(checkTimes.get(), is(1));
+                        });
```

Comment:
Maybe also verify that:
* `checkTimes` increase again if there's another `processResourceRequirements` after the first `triggerNonPeriodicScheduledTasks`
* `checkTimes` does not increase when there's no events between two `triggerNonPeriodicScheduledTasks`